airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -363,3 +340,6 @@ def _get_message(self, record_data_or_message: Union[StreamData, AirbyteMessage]
     @property
     def message_repository(self) -> Union[None, MessageRepository]:
         return None
+
+    def get_slice_logger(self) -> SliceLogger:
```

Comment:
why output a new obj every time instead of use an instance var?

Reply:
good call. updated to use an instance field